tests/run.sh

```diff
@@ -204,6 +204,12 @@ for f in $tests; do
     # Send the test file to awesome.
     cat $f | DISPLAY=$D "$AWESOME_CLIENT" 2>&1
 
+    # Kill awesome after 1 minute (e.g. with errors during test setup).
+    (sleep 60; if [ "$(ps -o comm= $awesome_pid)" = "${AWESOME##*/}" ]; then
+        echo "Killing (stale?!) awesome (PID $awesome_pid) after 10 seconds."
```

Comment:
Oh and apropos what this commit actually does: I tried to fix this with e18bece3df8911102c92f60fc06fa5da62438c35, but I guess the problem was re-introduced since then? Am I understanding this correctly? If so, your fix (while ugly) should fix this once and for all.